app/src/main/java/org/dhis2/usescases/teiDashboard/DashboardViewModel.kt

```diff
@@ -277,4 +281,40 @@ class DashboardViewModel(
     fun onNavigationItemSelected(itemId: TEIDashboardItems) {
         _navigationBarUIState.value = _navigationBarUIState.value.copy(selectedItem = itemId)
     }
+
+    fun checkIfTeiCanBeTransferred(
+        onResult: (Boolean) -> Unit
+    ) {
+        viewModelScope.launch(dispatcher.io()) {
+            try {
+                val canTransfer = repository.teiCanBeTransferred().blockingGet()
```

Comment:
What do you think about couple `blockingGet()` into the repository to avoid using RX dependencies in the ViewModel?. Yo can return just a boolean

Reply:
Done